tests/components/vacuum/test_init.py

```diff
@@ -42,33 +46,58 @@ async def test_state_services(
     hass: HomeAssistant, config_flow_fixture: None, service: str, expected_state: str
 ) -> None:
     """Test get vacuum service that affect state."""
+    mock_vacuum = MockVacuum(
+        name="Testing",
+        entity_id="vacuum.testing",
+    )
+    setup_test_component_platform(
+        hass,
+        DOMAIN,
+        [mock_vacuum],
+    )
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "test"}})
+    await hass.async_block_till_done()
 
-    entity0 = await create_entity(hass)
+    config_entry = MockConfigEntry(domain="test", data={})
```

Comment:
When we want to set up a platform from a config entry we need to do it like in this PR:

https://github.com/home-assistant/core/blob/4c344b6ccdb8d0b6e9c2e1769aa549f100865913/tests/components/notify/test_init.py#L67-L109

Reply:
Hows that?